enarx-keep-sgx-shim/src/handler.rs

```diff
@@ -182,4 +195,74 @@ impl<'a> Handler<'a> {
             ret
         }
     }
+
+    /// Do a readv() syscall
+    pub fn readv(&mut self) -> u64 {
+        let fd = self.aex.gpr.rdi;
+        let trusted = unsafe {
+            from_raw_parts_mut(self.aex.gpr.rsi as *mut Iovec, self.aex.gpr.rdx as usize)
+        };
+
+        // Add up total size of buffers.
+        let bufsize = trusted.iter().fold(0, |a, e| a + e.size);
+
+        // Allocate some unencrypted memory.
+        let size = bufsize + size_of::<Iovec>() * trusted.len();
+        let map = match self.ualloc(size as u64) {
+            Err(errno) => return errno.into_syscall(),
+            Ok(map) => unsafe { from_raw_parts_mut(map, size as usize) },
+        };
+
+        // Split allocated memory into that used by the iovec struct array and that used by its buffers.
+        let (uiovec, ubuffer) = map.split_at_mut(size_of::<Iovec>() * trusted.len());
+
+        // Convert the prefix from a byte slice into an iovec slice.
+        let (_, untrusted, _) = unsafe { uiovec.align_to_mut::<Iovec>() };
+        if untrusted.len() != trusted.len() {
+            self.attacked();
+        }
+
+        // Set pointers in unencrypted iovec slice to use the rest of the allocated memory.
+        // The offset is into the buffer area allocated immediately after the iovec struct
+        // array, measured in bytes.
+        let mut offset = 0;
+        for (t, mut u) in trusted.iter_mut().zip(untrusted.iter_mut()) {
+            u.base = ubuffer[offset..].as_mut_ptr();
+            u.size = t.size;
+            offset += t.size;
+        }
+
+        // Do the syscall; replace encrypted memory with unencrypted memory.
+        let ret = unsafe {
+            self.syscall(
+                SysCall::READV,
+                fd,
+                untrusted.as_ptr() as _,
+                trusted.len() as u64,
```

Comment:
```suggestion
                untrusted.len() as u64,
```

These values are the same. But I think this is slightly more readable.